### PR TITLE
Add server side sorting

### PR DESCRIPTION
**Pull request by Suvra19**
Added code to pass the sort column and direction to the `updateData` callback.

### CODE REVIEW

File: src/__tests__/components/Table.test.js [hunk 1]
```diff
@@ -234,4 +234,52 @@ describe('Table', () => {
...
+                    direction: 'ascending'
+                }
+                wrapper.setState({ pagination, sort });
+                instance.sortRows({ column: 'firstName' });
+
+                expect(updateDataSpy).toHaveBeenCalledTimes(1);
+                expect(updateDataSpy).toHaveBeenCalledWith({page: pagination.currentPage, sort: {...sort, direction: 'descending'}});
+            });
+
+            it('should call updateData with updated sort column and the default ascending direction', () => {
```
Comment (JonForest) [inline src/__tests__/components/Table.test.js]: It's not clear to me why the first test results in descending data, and the second in ascending data.
As near as I can tell, the set-up for both tests looks the same (i.e. the sort object both have the same initial `direction`).
Think that needs some explanation.

File: src/components/Table.js [hunk 2]
```diff
@@ -6,7 +6,7 @@ import Search from './Search';
 import Columns from './Columns';
 import Rows from './Rows';
 import Pagination from './Pagination';
-import { calculateRows, sortColumn, nextPage, previousPage, goToPage, expandRow, setInputtedPage } from '../actions/TableActions'
+import { calculateRows, sortColumn, nextPage, previousPage, goToPage, expandRow, changeSortFieldAndDirection } from '../actions/TableActions'
```
Comment (JonForest) [inline src/components/Table.js:9]: Surprised `setInputtedPage` is removed as it's not in any code changes below.  Is it no longer required?  Is it used anywhere?

Comment (JonForest) [reply src/components/Table.js:9]: Is `changeSortFieldAndDirection` your function or existing?
image

If your function, presumably it should be in this PR?

**Review by JonForest — CHANGES_REQUESTED** [review]
The build/index.js needs to not be included in this PR.

File: src/__tests__/components/Table.test.js (hunk 1 above)
Comment (Suvra19) [reply src/__tests__/components/Table.test.js]: I will change the wording a little. The first test basically tests that if the sort direction is reversed when the same column is requested to be sorted again. And the second one tests that when a new column is selected to be sorted, the new column is set as the sort column and direction is set to ascending by default.

File: src/components/Table.js (hunk 2 above)
Comment (Suvra19) [reply src/components/Table.js:9]: Yeah `setInputtedPage` is an unused import (even in the source master). `changeSortFieldAndDirection` is an existing function that is used to update the sort column and direction in the state. The row sorting is performed after that by a separate method. For server side sorting, we need the sort column and direction update  logic to remain the same so I have reused the method.

**Review by JonForest — APPROVED** [review]
LGTM

Note, you'll need to update the Releases and then update the package.json in the AQ codebase for this to be pulled in. 
I suspect you know that however.

**Pull request merged**